src/EnergyPlus/Boilers.cc

```diff
@@ -900,15 +895,15 @@ void BoilerSpecs::CalcBoilerModel(EnergyPlusData &state,
     Real64 EffCurveOutput = 1.0;                                 // Output of boiler efficiency curve
 
     // calculate normalized efficiency based on curve object type
-    if (this->EfficiencyCurvePtr > 0) {
-        if (state.dataCurveManager->PerfCurve(this->EfficiencyCurvePtr)->numDims == 2) {
+    if (this->EfficiencyCurve != nullptr) {
+        if (this->EfficiencyCurve->numDims == 2) {
             if (this->CurveTempMode == TempMode::ENTERINGBOILERTEMP) {
-                EffCurveOutput = Curve::CurveValue(state, this->EfficiencyCurvePtr, this->BoilerPLR, state.dataLoopNodes->Node(BoilerInletNode).Temp);
+                EffCurveOutput = this->EfficiencyCurve->value(state, this->BoilerPLR, state.dataLoopNodes->Node(BoilerInletNode).Temp);
```

Comment:
New object/method-based curve access pattern. Straightforward search and replace.